src/hellpers/main.hellpers.js

```diff
@@ -6,9 +6,64 @@ const getRequestBody = req => new Promise((res, rej) => {
     .on('data', chunk => buffer += chunk)
     .on('error', err => rej(err))
     .on('end', () => res(buffer))
-})
+});
+
+const getResponseBody = data => new Promise(res => {
+
+  const { searchParams } = URL.incomingRoute;
+  const body = [];
+
+  searchParams.forEach((val, key) => {
+    data.forEach(el => {
+      const arr = Array.isArray(el[key]) ? el[key] : [el[key]];
+      if (arr.includes(val)) body.push(el);
+    })
+  })
+
+  return res(body);
+});
+
+const parseRoute = req => new Promise(res => {
+
+  class IncomingRoute extends URL {
+    constructor(req) {
+      super(`http://${req.headers.host}${req.url}`)
+      this.method = req.method;
+      this.setCorrectData();
+    }
 
+    setCorrectData() {
+      const { pathname, method, search } = this;
+      const pathArr = pathname.replace(/\/$/, '').slice(1).split('/');
+      let name = '';
+
+      if (method === 'GET' && pathArr.length === 2 && !search) this.getAll = true;
+
+      for (let i = 0; i < pathArr.length; i++) {
+        if (i <= 1) name += `/${pathArr[i]}`;
+        if (i > 1) {
+          const val = Number.isNaN(+pathArr[i]) ? 'categories' : 'id';
```

Comment:
It's not really generic approach. Probably it would be better to receive something similar to express route string (with /:id etc.) and try to match url for handler and parse path-params and query-params. But it would feet well for the purpose of this task, this won't affect total mark for this Homework.